### ISSUE TITLE
realm.deleteAll() does not delete stale data

### ISSUE THREAD

**Issue by hyb175** [realm/realm-js#1268]
## Goals

When I remove a model and update the schema, I would love to clean up the data as well.

## Expected Results

`realm.deleteAll()` would remove all data, no matter if it is in the schema or not.

## Actual Results

The data of the deleted model still persists. 

## Steps to Reproduce

If I had a Car model, but the new schema got rid of it. Even after I call `realm.deleteAll()`, the stale data from the old Car model still exists.

## Version of Realm and Tooling

- Realm JS SDK Version: 1.3.1
- Node or React Native: react-native 0.44.0
- Which debugger for React Native: None

Right now I am deleting the `.realm` file before initializing to get rid of the stale data, but it seems to me that `realm.deleteAll()` should mean a clean slate with the new schema and without any stale data like this.

**Comment by kneth** [comment]
`deleteAll()` only operates on what's in the schema. Deletion should happen during schema migration. Can you share the migration code?

**Comment by hyb175** [comment]
My migration code looks something like this.
```javascript
export default function modelRename(oldRealm, newRealm) {
  const users = oldRealm.objects('User');
  users.forEach((user) => {
    user.trucks.forEach((truck) => { // trucks used to be of model `Car`, now they are model `Truck`
      newRealm.create('Truck', truckMapper(truck));
    });
  });
}
```

I am not sure when to delete the old data. I have tried the following:
```javascript
export default function modelRename(oldRealm, newRealm) {
  const users = oldRealm.objects('User');
  users.forEach((user) => {
    user.trucks.forEach((truck) => { // trucks used to be of model `Car`, now they are model `Truck`
      newRealm.create('Truck', truckMapper(truck));
      oldRealm.delete(truck);
    });
  });
}
```
But since `oldRealm` is read only, I cannot delete it like this. 
I have also tried using `newRealm.delete(truck)`. It would complain about not knowing what `Car` model is.
I had to resolve to delete the `.realm` file for now to get rid of stale data, but I feel like that is not idea. I was hoping `deleteAll` would delete everything and reinitialize the database with the right models. Can I ask why `deleteAll` would only operate on the current schema instead of wipe everything clean?

**Comment by hyb175** [comment]
I see https://github.com/realm/realm-js/pull/1269 is in the works which should make deleting the Realm file easier. Is this gonna be the way to handle the problem that I am facing?

**Comment by kneth** [comment]
@hyb175 `deleteRealm()` will delete the Realm file. That means that all objects will be deleted. As you can see from #1269, we have a few issues with iOS builds but I hope we have it ready soon.

**Comment by kneth** [comment]
@hyb175 We don't expose a method to delete a class (and it's objects) during migration. In that respect, your case is similar to #573.

**Comment by hyb175** [comment]
@kneth and from the status of that feature request, it has been picked up and worked on I assume?

**Comment by kneth** [comment]
We have a workaround for the iOS issues seen in #1269, so I hope we will be able to merge and release it soon.

**Comment by hyb175** [comment]
@kneth Just to clarify, the way to solve my issue, which is similar to https://github.com/realm/realm-js/issues/573 is to delete the Realm file for now? Is there gonna be a solution without the need to remove the file? From the status of https://github.com/realm/realm-js/issues/573, it looks like it was in backlog and got picked up recently.

**Comment by kneth** [comment]
Currently, `deleteRealm()` is your best option. But yes, @ashwinphatak is working on #573.

**Comment by ashwinphatak** [comment]
Merged 532f517, which exposes a `deleteModel` method to be used during migrations.

**Comment by hyb175** [comment]
Thank you! @ashwinphatak
